src/fvm_multicell.hpp

```diff
@@ -85,8 +85,56 @@ class fvm_multicell {
         return (this->*h.first)[h.second];
     }
 
-    /// integrate all cell state forward in time
-    void advance(double dt);
+    /// Initialize state prior to a sequence of integration steps.
+    void start_integration(value_type tfinal, value_type dt_max) {
+        EXPECTS(tfinal>t_);
+        EXPECTS(dt_max>0);
+
+        tfinal_ = tfinal;
+        dt_max_ = dt_max;
+        integration_running_ = true;
+
+        // TODO: Placeholder; construct backend event delivery
+        // data structure from events added.
+        EXPECTS(events_.empty());
+        for (auto& ev: staged_events_) {
+            EXPECTS(ev.time<tfinal);
+            events_.push(std::move(ev));
+        }
+
+        staged_events_.clear();
+    }
+
+    /// Advance one integration step.
+    void step_integration();
+
+    /// Query integration completion state.
+    bool integration_complete() const {
+        return !integration_running_;
+    }
+
+    /// Query per-cell time state. (Placeholders)
+    value_type time(size_type cell_index) const {
+        return t_;
+    }
+
+    value_type max_time() const {
+        return t_;
+    }
+
+    bool state_synchronized() const {
+        return true;
+    }
+
+    /// Add an event for processing in next integration stage.
```

Comment:
The assumption here is no: all the events for one 'big' interval and no others are expected to be passed on to the lowered cell. It may be possible to relax this assumption, but I wouldn't do so unless there is a clear gain.

The `cell_group` is responsible for queuing, and holds the priority queue data structure. The lowered implementation now has responsibility for delivering, and the API is meant to enforce a sequence of:
1. `cell_group` provides all the events, associated with their gid.
2. `cell_group` instructs lowered implementation to bulk-load/prepare the provided events for delivery within the integration interval.
3. The lowered implementation uses a pop/top-only queue for handling the discrete deliveries internally over the course of processing an integration interval.